ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/recovery/RecoverySocketUnitTest.java

```diff
@@ -143,6 +142,35 @@ public void socketScanVerbose() throws Exception {
         }
     }
 
+    @Test
+    public void socketScanTerminateAbruptly() throws Exception {
+        try (Socket connectorSocket = getSocket()) {
+            PrintWriter toServer = new PrintWriter(new OutputStreamWriter(connectorSocket.getOutputStream(), StandardCharsets.UTF_8));
+            toServer.println(RecoveryDriver.SCAN);
+            toServer.flush();
+            recoveryManager.terminate();
```

Comment:
I wonder if byteman could ensure that this terminated occurs at the right time to detect that the scan has started as is being interrupted?

Also, should we be checking the outout from the read side to make sure the scan took place?

Reply:
The terminate needs to occur before the `SCAN` command returns back any value. Otherwise the test would not verifies about the abrupt termination.

Sure Byteman could be used. If you prefer that I can try to add some byteman test for this.
I personally prefer here the standard "blackbox" test which verifies if the termination works or has some issue.

Reply:
I will close this discussion to move it closer to where I would most like to explore